contrib/drivers/ydb/ydb.go

```diff
@@ -0,0 +1,148 @@
+		source = config.Link
+		// Custom changing the schema in runtime.
+		if config.Name != "" {
+			source, _ = gregex.ReplaceString(`/([\w\.\-]+)+`, "/"+config.Name, source)
+		}
+	} else {
+		source = fmt.Sprintf(
+			"%s:%s@%s(%s:%s)/%s?charset=%s",
+			config.User, config.Pass, config.Protocol, config.Host, config.Port, config.Name, config.Charset,
+		)
+		if config.Timezone != "" {
+			if strings.Contains(config.Timezone, "/") {
+				config.Timezone = url.QueryEscape(config.Timezone)
+			}
+			source = fmt.Sprintf("%s&loc=%s", source, config.Timezone)
+		}
+		if config.Extra != "" {
+			source = fmt.Sprintf("%s&%s", source, config.Extra)
+		}
+	}
+	if db, err = sql.Open(underlyingDriverName, source); err != nil {
+		err = gerror.WrapCodef(
+			gcode.CodeDbOperationError, err,
+			`sql.Open failed for driver "%s" by source "%s"`, underlyingDriverName, source,
+		)
+		return nil, err
+	}
+	return
+}
+
+// Tables retrieves and returns the tables of current schema.
+// It's mainly used in cli tool chain for automatically generating the models.
+func (d *Driver) Tables(ctx context.Context, schema ...string) (tables []string, err error) {
+	var result gdb.Result
+	link, err := d.SlaveLink(schema...)
+	if err != nil {
+		return nil, err
+	}
+	result, err = d.DoSelect(ctx, link, `SHOW TABLES`)
+	if err != nil {
+		return
+	}
+	for _, m := range result {
+		for _, v := range m {
+			tables = append(tables, v.String())
+		}
+	}
+	return
+}
+
+// TableFields retrieves and returns the fields' information of specified table of current
+// schema.
+//
+// The parameter `link` is optional, if given nil it automatically retrieves a raw sql connection
+// as its link to proceed necessary sql query.
+//
+// Note that it returns a map containing the field name and its corresponding fields.
+// As a map is unsorted, the TableField struct has a "Index" field marks its sequence in
+// the fields.
+//
+// It's using cache feature to enhance the performance, which is never expired util the
+// process restarts.
+func (d *Driver) TableFields(
+	ctx context.Context, table string, schema ...string,
+) (fields map[string]*gdb.TableField, err error) {
+	var (
+		result    gdb.Result
+		link      gdb.Link
+		useSchema = gutil.GetOrDefaultStr(d.GetSchema(), schema...)
+	)
+	if link, err = d.SlaveLink(useSchema); err != nil {
+		return nil, err
+	}
+	result, err = d.DoSelect(
+		ctx, link,
+		fmt.Sprintf(`SHOW FULL COLUMNS FROM %s`, d.QuoteWord(table)),
+	)
+	if err != nil {
+		return nil, err
+	}
+	fields = make(map[string]*gdb.TableField)
+	for i, m := range result {
+		fields[m["Field"].String()] = &gdb.TableField{
+			Index:   i,
+			Name:    m["Field"].String(),
+			Type:    m["Type"].String(),
+			Null:    m["Null"].Bool(),
+			Key:     m["Key"].String(),
+			Default: m["Default"].Val(),
+			Extra:   m["Extra"].String(),
+			Comment: m["Comment"].String(),
+		}
+	}
+	return fields, nil
+}
```

Comment:
Please add unit testing cases for new driver `ydb`. You can refer to other drivers like `sqlite` https://github.com/gogf/gf/tree/master/contrib/drivers/sqlite